trite-osprey/src/List.js

```diff
@@ -0,0 +1,219 @@
+import React from 'react';
+import Cart from './Cart';
+
+class List extends React.Component{
+
+  constructor(props) {
+    super(props)
+    this.state = {
```

Comment:
This is a huuuuuge amount of state. I'd try and reduce this as much as possible.